clb.cabal

```diff
@@ -57,41 +58,37 @@ library
     Clb.Tx
 
   build-depends:
-    , base                        ^>=4.16.4.0
+    , base                        ^>=4.18.1.0
     , base16-bytestring           ^>=1.0.2.0
-    , bytestring                  ^>=0.11.4.0
-    , containers                  ^>=0.6.5.1
+    , bytestring
+    , containers                  ^>=0.6.7
     , data-default                ^>=0.7.1.1
-    , effectful                   ^>=2.3.0.0
-    , effectful-core              ^>=2.3.0.1
-    , effectful-plugin            ^>=1.1.0.2
-    , effectful-th                ^>=1.0.0.1
     , lens                        ^>=5.2.3
-    , mtl                         ^>=2.2.2
+    , mtl                         ^>=2.3.1
     , prettyprinter               ^>=1.7.1
     , text                        ^>=2.0.2
-    , time                        ^>=1.11.1.1
-    , transformers                ^>=0.5.6.2
+    , time                        ^>=1.12.2
+    , transformers                ^>=0.6.1.0
 
   build-depends:
-    , cardano-api:{cardano-api, internal}                 ^>=8.37.1.0
-    , cardano-binary                                      ^>=1.7.1.0
-    , cardano-crypto-class                                ^>=2.1.4.0
-    , cardano-ledger-alonzo                               ^>=1.5.1.0
-    , cardano-ledger-api                                  ^>=1.7.0.1
-    , cardano-ledger-babbage                              ^>=1.5.1.0
-    , cardano-ledger-conway                               ^>=1.11.0.0
-    , cardano-ledger-core:{cardano-ledger-core, testlib}  ^>=1.9.0.0
-    , cardano-ledger-mary                                 ^>=1.4.0.0
-    , cardano-ledger-pretty                               ^>=1.3.3.1
-    , cardano-ledger-shelley                              ^>=1.8.0.0
-    , cardano-slotting                                    ^>=0.1.2.0
-    , ouroboros-consensus-cardano                         ^>=0.12
-    , plutus-ledger-api                                   ^>=1.15.0.0
-    , plutus-tx                                           ^>=1.15.0.0
+    , cardano-api:{cardano-api, internal}                 ==8.38.0.0
```

Comment:
Why pin versions if there is freeze file?

Reply:
We can remove them, but they might still be useful if we use `^>=`. I think `cabal outdated` command is capable of telling you which dependencies are obsolete and I wonder how its behavior is affected by having those limits in a cabal file. Agreed we need to avoid using `==` in cabal files. Another cabal issue -- `cabal gen-bounds` fails to me with plan building error, I don't know why:

> [euonymos@nixos:~/src/mlabs/clb-cek]$ cabal gen-bounds
> Resolving dependencies...
> Error: cabal: Could not resolve dependencies:
> [__0] trying: clb-0.0.1 (user goal)
> [__1] unknown package: plutus-tx (dependency of clb)
> [__1] fail (backjumping, conflict set: clb, plutus-tx)
> After searching the rest of the dependency tree exhaustively, these were the
> goals I've had most trouble fulfilling: clb, plutus-tx

Reply:
Why bounds would be needed anyway? They lead to compilation problems and thus more PRs and code synchronization.

Right now they do not work, at least `== 9.9.9.9` does not, so I still use `allow-newer`.

If some dep needs to be version enforced, usually one `>=` constraint might fix that. It is simpler to debug to me, cuz you know causal relationships. If one want to check exact versions, they could just take a look at freeze.

Reply:
What's exactly not working for you? It's indeed 9.9.9.9 and compiles well at my end.

Reply:
You right there was just import missing.